### PR TITLE
chore: add labeler to add sig labels

### PR DESCRIPTION
**Pull request by hi-rustin**
Signed-off-by: hi-rustin <rustin.liu@gmail.com>

### What problem does this PR solve?

Issue Number: None.

Problem Summary:

add labeler to add sig labels.

### What is changed and how it works?
None

What's Changed:

### Related changes

None

### Check List 

Tests 

- No code

Side effects
None

### Release note 
- No release note

### CODE REVIEW

**Comment by hi-rustin** [comment]
/cc @BusyJay @breeswish @zhangjinpeng1987 

There may be a lot of problems as I am not very familiar with the tikv code base. If there are problems please fix them by modifying my pr directly. If you guys don't think it's necessary to add the tool, please feel free to close the PR.

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  - "cmd/*"
+  - "components/backup/*"
+  - "components/collections/*"
+  - "components/engine_*/*"
+  - "components/error_code/src/engine.rs"
+  - "components/error_code/src/lib.rs"
+  - "components/error_code/src/raftstore.rs"
+  - "components/error_code/src/sst_importer.rs"
+  - "components/error_code/src/storage.rs"
+  - "components/error_code/src/bin.rs"
```
Comment (breeswish) [inline .github/labeler.yml:11]: Maybe we can simply leave error code alone, since the code may be frequently changed.

Comment (hi-rustin) [reply .github/labeler.yml:11]: Got it.

**Review by nrc — APPROVED** [review]
I added a commit moving some modules to sig-txn. I think there are some 'general' modules in eig-engine and sig-coprocessor, but otherwise lgtm.

**Comment by hi-rustin** [comment]
@breeswish @yiwu-arbug  @BusyJay  ping~

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  - "src/server/*"
+  - "src/config.rs"
+  - "src/lib.rs"
+  - "src/read_pool.rs"
+
+sig/raft:
+  - "components/batch-system/*"
+  - "components/cdc/*"
+  - "components/configuration/*"
+  - "components/encryption/*"
```
Comment (yiwu-arbug) [inline .github/labeler.yml:28]: `components/encryption` should be under sig/engine

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  - "components/test_raftstore/*"
+  - "components/test_sst_importer/*"
+  - "components/tikv_util/*"
+  - "etc/*"
+  - "metrics/*"
+  - "src/import/*"
+  - "src/server/*"
+  - "src/config.rs"
+  - "src/lib.rs"
+  - "src/read_pool.rs"
```
Comment (yiwu-arbug) [inline .github/labeler.yml:22]: The following does not belong to sig/engine:
```
cmd/*
components/backup/*
components/collections/*
components/external_storage/*
components/keys/*
components/raftstore/*
components/sst_importer/*
components/test_raftstore/*
components/test_sst_importer/*
components/tikv_util/*
etc/*
metrics/*
src/import/*
src/server/*
src/config.rs
src/lib.rs
src/read_pool.rs
```

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+sig/raft:
+  - "components/batch-system/*"
+  - "components/cdc/*"
+  - "components/configuration/*"
+  - "components/encryption/*"
+  - "components/into_other/*"
+  - "components/resolved_ts/*"
+  - "components/security/*"
```
Comment (yiwu-arbug) [inline .github/labeler.yml:31]: `components/security` can move to sig/engine.

File: .github/labeler.yml (hunk 2 above)
Comment (hi-rustin) [reply .github/labeler.yml:28]: Can you modify my pr directly? I'm not very familiar with the tikv codebase.

File: .github/labeler.yml (hunk 3 above)
Comment (yiwu-arbug) [reply .github/labeler.yml:22]: @hi-rustin some of these directories don't related to a particular SIG. How can we handle them?

Comment (hi-rustin) [reply .github/labeler.yml:22]: If it involves more than one sig we can add more than one sig labels, or if it is absolutely impossible to divide the sig, we can remove it.

**Pull request closed without merging**